server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -465,38 +467,31 @@ public void onFailure(Throwable t)
             {
               openConnections.getAndDecrement();
               if (future.isCancelled()) {
```

Comment:
Please extract the whole block inside `if (future.isCancelled()) { ... }` as a method. There is too much nesting now (and will be even more, when `checkRunnable` is scheduled from inside `cancelRunnable`).